library/std/src/sys/uefi/args.rs

```diff
@@ -0,0 +1,152 @@
+use r_efi::protocols::loaded_image;
+
+use crate::env::current_exe;
+use crate::ffi::OsString;
+use crate::fmt;
+use crate::iter::Iterator;
+use crate::mem::size_of;
+use crate::sys::uefi::helpers;
+use crate::vec;
+
+pub struct Args {
+    parsed_args_list: vec::IntoIter<OsString>,
+}
+
+pub fn args() -> Args {
+    // SAFETY: Each loaded image has an image handle that supports EFI_LOADED_IMAGE_PROTOCOL
+    let protocol =
+        helpers::image_handle_protocol::<loaded_image::Protocol>(loaded_image::PROTOCOL_GUID)
+            .unwrap();
```

Comment:
When does this unwrap() happen? It doesn't look like we currently document any panics for the args_os iterator (which I believe this is implementing?)

I think at minimum it would be nice to add a comment and/or make this expect("...") with some short description.

Reply:
Well, the unwrap actually will never happen. As the SAFETY comment states, every image handle implements this protocol. Maybe I need to more clear?

Reply:
SAFETY usually refers to something unsafe (e.g., dereferencing a pointer). I would just drop "SAFETY:" if this is expected to describe the unwrap, perhaps adding "unwrap() shouldn't fail, each loaded ...".